File: 3-Business/3-Modules/lm.Comol.Core.BaseModules/Repository/Domain/dtoFileFolder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace lm.Comol.Core.BaseModules.Repository
{
    [Serializable]
    public class dtoFileFolder
    {
        public long Id {get;set;}
        public String Name {get;set;}
        public Boolean isVisible {get;set;}
        public List<dtoFileFolder> SubFolders {get;set;}
        public List<dtoGenericFile> Files {get;set;}
        public Boolean Selected {get;set;}
        public Boolean Selectable {get;set;}
        public Boolean HasFiles
        {
            get {
                return Files.Any() || SubFolders.Where(f => f.HasFiles).Any();
            }
        
        }

        public dtoFileFolder(){
            Selected = false;
            Files = new List<dtoGenericFile>();
            SubFolders = new List<dtoFileFolder>();
        }

        //Sub New(ByVal oId As Long, ByVal oName As String, ByVal oVisible As Boolean)
        //    Me._ID = oId
        //    Me._Name = oName
        //    Me._isVisible = oVisible
        //    Me._Selected = False
        //    Me._Files = New List(Of dtoGenericFile)
        //    Me._Folder = New List(Of dtoFileFolder)
        //End Sub
        //Sub New(ByVal oId As Long, ByVal oName As String, ByVal oVisible As Boolean, ByVal oSelected As Boolean)
        //    Me._ID = oId
        //    Me._Name = oName
        //    Me._isVisible = oVisible
        //    Me._Selected = oSelected
        //    Me._Files = New List(Of dtoGenericFile)
        //    Me._Folder = New List(Of dtoFileFolder)
        //End Sub
    }

    [Serializable]
    public class dtoGenericFile
    {
        public long Id {get;set;}
        public String Name {get;set;}
        public String Extension {get;set;}
        public String DisplayName {get{return Name+ Extension;}}
        public Boolean isVisible {get;set;}
        public dtoFileFolder Folder {get;set;}
        public Boolean Selected {get;set;}
        public Boolean Selectable {get;set;}

        public dtoGenericFile(){
            Selected = false;
            Selectable = true;
        }
    }
}


    //    Sub New()
    //        _Selected = False
    //        _Selectable = True
    //    End Sub
    //    Sub New(ByVal oId As Long, ByVal oName As String, ByVal oExtension As String, ByVal oVisible As Boolean)
    //        Me._ID = oId
    //        Me._Name = oName
    //        Me._Extension = oExtension
    //        Me._isVisible = oVisible
    //        _Selected = False
    //        _Selectable = True
    //    End Sub
    //    Sub New(ByVal oId As Long, ByVal oName As String, ByVal oExtension As String, ByVal oVisible As Boolean, ByVal oSelected As Boolean)
    //        Me._ID = oId
    //        Me._Name = oName
    //        Me._Extension = oExtension
    //        Me._isVisible = oVisible
    //        _Selected = oSelected
    //        _Selectable = True
    //    End Sub
    //    'Sub New(ByVal oId As Long, ByVal oName As String, ByVal oVisible As Boolean, ByVal oFolder As dtoFileFolder)
    //    '    Me._ID = oId
    //    '    Me._Name = oName
    //    '    Me._isVisible = oVisible
    //    '    Me._Folder = oFolder
    //    'End Sub

    //End Class
        

File: 3-Business/3-Modules/lm.Comol.Modules.Base/Domain/Core/dtoFileFolder.vb
Namespace lm.Comol.Modules.Base.DomainModel
    <Serializable(), CLSCompliant(True)> Public Class dtoFileFolder

#Region "Private Property"
        Private _ID As Long
        Private _Name As String
        Private _isVisible As Boolean
        Private _Files As List(Of dtoGenericFile)
        Private _Folder As List(Of dtoFileFolder)
        Private _Selected As Boolean
        Private _Selectable As Boolean
#End Region

#Region "Public Property"
        Public Property ID() As Long
            Get
                Return _ID
            End Get
            Set(ByVal value As Long)
                _ID = value
            End Set
        End Property
        Public Property Name() As String
            Get
                Return _Name
            End Get
            Set(ByVal value As String)
                _Name = value
            End Set
        End Property
        Public Property isVisible() As Boolean
            Get
                Return _isVisible
            End Get
            Set(ByVal value As Boolean)
                _isVisible = value
            End Set
        End Property
        Public Property SubFolders() As List(Of dtoFileFolder)
            Get
                Return _Folder
            End Get
            Set(ByVal value As List(Of dtoFileFolder))
                _Folder = value
            End Set
        End Property
        Public Property Files() As List(Of dtoGenericFile)
            Get
                Return _Files
            End Get
            Set(ByVal value As List(Of dtoGenericFile))
                _Files = value
            End Set
        End Property
        Public Property Selected() As Boolean
            Get
                Return _Selected
            End Get
            Set(ByVal value As Boolean)
                _Selected = value
            End Set
        End Property
        Public Property Selectable() As Boolean
            Get
                Return _Selectable
            End Get
            Set(ByVal value As Boolean)
                _Selectable = value
            End Set
        End Property
#End Region

        Sub New()
            Me._Selected = False
            Me._Files = New List(Of dtoGenericFile)
            Me._Folder = New List(Of dtoFileFolder)
        End Sub
        Sub New(ByVal oId As Long, ByVal oName As String, ByVal oVisible As Boolean)
            Me._ID = oId
            Me._Name = oName
            Me._isVisible = oVisible
            Me._Selected = False
            Me._Files = New List(Of dtoGenericFile)
            Me._Folder = New List(Of dtoFileFolder)
        End Sub
        Sub New(ByVal oId As Long, ByVal oName As String, ByVal oVisible As Boolean, ByVal oSelected As Boolean)
            Me._ID = oId
            Me._Name = oName
            Me._isVisible = oVisible
            Me._Selected = oSelected
            Me._Files = New List(Of dtoGenericFile)
            Me._Folder = New List(Of dtoFileFolder)
        End Sub
    End Class
End Namespace